assets/js/util/insufficient-permissions-error-description.js

```diff
@@ -26,73 +26,164 @@ import { __, sprintf } from '@wordpress/i18n';
+		default:
+			messages = [];
+			break;
+	}
+
+	if ( messages.length === 0 ) {
+		messages = emptyError( name );
+	}
+
+	messages = [ ...messages, ...userLoginErrors( owner ) ];
+
+	//  Create a full sentence separated by full stops "." and a space to separate 2 messages, if more than one message
+	const sentence = messages.join( '. ' );
+
+	/**
+	 * If the message contains less than 2 sentences return the sentence as it is, otherwise add a full stop at the
+	 * end of the sentence, in order to follow the UX guidelines.
+	 *
+	 * "text with 1 or 2 sentences should not end in a full stop".
+	 *
+	 * @since n.e.x.t
+	 * @see https://github.com/google/site-kit-wp/issues/4160
+	 */
+	return messages.length <= 2 ? sentence : `${ sentence }.`;
+}
 
-	if ( 'analytics' === slug ) {
-		if ( error.match( /account/i ) ) {
-			message = __(
+/**
+ * Gets a list of error messages for the Analytics module.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} error The error message provided.
+ * @return {string[]} An array of string with a list of errors messages.
+ */
+function analyticsError( error ) {
+	if ( error.match( /account/i ) ) {
+		return [
+			__(
 				`Your Google account does not have sufficient permissions for this Analytics account, so you won't be able to see stats from it on the Site Kit dashboard`,
 				'google-site-kit'
-			);
-		} else if ( error.match( /property/i ) ) {
-			message = __(
-				`Your Google account does not have sufficient permissions for this Analytics property, so you won't be able to see stats from it on the Site Kit dashboard`,
+			),
+		];
+	}
+
+	if ( error.match( /property/i ) ) {
+		return [
+			__(
+				`Your Google account does not have sufficient permissions for this Analytics view, so you won't be able to see stats from it on the Site Kit dashboard`,
 				'google-site-kit'
-			);
-		} else if ( error.match( /view/i ) ) {
-			message = __(
+			),
+		];
+	}
+
+	if ( error.match( /view/i ) ) {
+		return [
+			__(
 				`Your Google account does not have sufficient permissions for this Analytics view, so you won't be able to see stats from it on the Site Kit dashboard`,
 				'google-site-kit'
-			);
-		}
-	} else if ( 'search-console' === slug ) {
-		message = __(
+			),
+		];
+	}
+
+	return [];
+}
+
+/**
+ * Gets a list of errors for the search console module.
+ *
+ * @since n.e.x.t
+ *
+ * @return {string[]} An array of string with a list of errors messages.
+ */
+function searchConsoleErrors() {
+	return [
+		__(
 			`Your Google account does not have sufficient permissions for this Search Console property, so you won't be able to see stats from it on the Site Kit dashboard`,
 			'google-site-kit'
-		);
-	}
+		),
+	];
+}
 
-	if ( ! message ) {
-		message = sprintf(
+/**
+ * Gets the errors the error messages are empty.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string} moduleName The name of the module.
+ * @return {string[]} An array of string with a list of errors messages.
+ */
+function emptyError( moduleName ) {
```

Comment:
Same here. No need to create a function that consists only from a return statement that returns a static string. Let's incorporate it back into the parent function.